pkg/monitoring/domainstats/downwardmetrics/hostmetrics.go

```diff
@@ -77,10 +123,59 @@ func (h *hostMetricsCollector) Collect() (metrics []api.Metric) {
 
 func defaultHostMetricsCollector() *hostMetricsCollector {
 	return &hostMetricsCollector{
-		procCPUInfo: "/proc/cpuinfo",
-		procStat:    "/proc/stat",
-		procMemInfo: "/proc/meminfo",
-		procVMStat:  "/proc/vmstat",
-		pageSize:    os.Getpagesize(),
+		procPath: "/proc",
+		sysPath:  "/sys",
+		pageSize: os.Getpagesize(),
+	}
+}
+
+func derefOrZero(val *uint64) uint64 {
+	if val == nil {
+		return 0
+	}
+	return *val
+}
+
+type vmStat struct {
+	pswpin  uint64
+	pswpout uint64
+}
```

Comment:
nit: maybe move this type declaration to the top of the file

Reply:
done